### PR TITLE
Update scRNA tutorial

### LINKED ISSUE #153: Update basic clustering tutorial to be in sync with scanpy 1.10
Scanpy now features a cut down version of the basic plotting tutorial hosted here, which shows all the steps that only rely on scanpy.

https://scanpy.readthedocs.io/en/stable/tutorials/basics/clustering.html

While updating this, I made some modification that take advantage of new scanpy features. We should update the version here to be in sync.

### PR DESCRIPTION
**Pull request by flying-sheep**
Fixes #153

## basic clustering tutorial notebook
- Seems like the manual annotation is already broken, so I removed it.
  If you have an idea how we can keep it stable, we could do that instead, but I couldn’t figure out which cluster assignments actually work.
- Since executablebooks/MyST-NB#596 hasn’t been merged for ages, I’m installing it from git for now. It’s a good way to reference not-too-stable numbers from text without having to manually keep an eye on things

## Other notebooks
- fixes scipy compat by changing `csr_matrix.`{`A`→`toarray()`}
- fixes the backed mode tutorial’s `dotplot`, which used `dendrogram=True`: https://github.com/scverse/scanpy/issues/3199

## Conda setup
Some of the conda setup does no longer work, namely `use-only-tar-bz2`, so I needed to disable it. Let’s fully fix the setup in another PR, this PR makes it work again which was hard enough.

We should probably go away from conda in the first place. Installing pip packages into a conda environment is an undefined operation, all version bounds for the conda packages become meaningless when doing that. `uv` could be enough, otherwise we could go pixi or so.

### CODE REVIEW

**Comment by grst** [comment]
> Seems like the manual annotation is already broken, so I removed it.
If you have an idea how we can keep it stable, we could do that instead, but I couldn’t figure out which cluster assignments actually work.

In https://github.com/scverse/scanpy/issues/2014, we figured out that setting
```
export NUMBA_CPU_NAME=generic
```
makes the clustering results more stable (at least in the conditions I tested there) by only relying on basic CPU features. Would still need to redo it once but then hopefully it wouldn't change anymore in the future.

File: docs/patched-deps.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# fix from here: https://github.com/executablebooks/MyST-NB/pull/597
```
Comment (grst) [inline docs/patched-deps.txt:1]: Why is this necessary on top of the pyproject.toml?

**Review by grst — COMMENTED** [review]
LGTM! Thank you for suffering for us! 

I would be in favor of giving it a shot to try out the reproducible clustering with the numba variable as manual clustering is still a commonly used approach.

**Review by grst — APPROVED** [review]

**Comment by grst** [comment]
And not sure why we went with conda in the first place, maybe because of compatibility with mybinder?

File: docs/patched-deps.txt (hunk 1 above)
Comment (flying-sheep) [reply docs/patched-deps.txt:1]: because
- you can’t directly specify deps in .readthedocs.yml, only through referencing paths
- you can’t specify direct dependencies in standard metadata, only tool-specific metadata (e.g. if we’d use hatch or poetry to create our envs, we could use that)

Comment (grst) [reply docs/patched-deps.txt:1]: It certainly works to install the package + extra dependencies listed in pyproject.toml
```
python:
  install:
    - method: pip
      path: .
      extra_requirements:
        - docs
```

Not that its super important if it's solved one way or another, was mostly curious.

**Comment by flying-sheep** [comment]
let’s restore that in another PR, I don’t feel like going through all clusters and assigning labels manually.
